# GitLab NGINX Patch: Allow to configure automounting ServiceAccount token with global
# chart configuration.
diff --git a/charts/nginx-ingress/templates/_helpers.tpl b/charts/nginx-ingress/templates/_helpers.tpl
index a1e4ecffa..637e82e2d 100644
--- a/charts/nginx-ingress/templates/_helpers.tpl
+++ b/charts/nginx-ingress/templates/_helpers.tpl
@@ -214,6 +214,34 @@ Create the name of the default backend service account to use
 {{- end -}}
 {{- end -}}
 
+{{/*
+Set if the default ServiceAccount token should be mounted by Kubernetes or not.
+
+Default is 'true'
+*/}}
+{{- define "ingress-nginx.automountServiceAccountToken" -}}
+automountServiceAccountToken: {{ pluck "automountServiceAccountToken" .Values.serviceAccount .Values.global.serviceAccount | first }}
+{{- end -}}
+
+{{/*
+Set if the default ServiceAccount token should be mounted by Kubernetes or not.
+
+Default is 'true'
+*/}}
+{{- define "ingress-nginx.defaultBackend.automountServiceAccountToken" -}}
+automountServiceAccountToken: {{ pluck "automountServiceAccountToken" .Values.defaultBackend.serviceAccount .Values.global.serviceAccount | first }}
+{{- end -}}
+
+{{/*
+Set if the default ServiceAccount token should be mounted by Kubernetes or not.
+
+Default is 'true'
+*/}}
+{{- define "ingress-nginx.admissionWebhooks.patch.automountServiceAccountToken" -}}
+automountServiceAccountToken: {{ pluck "automountServiceAccountToken" .Values.admissionWebhooks.patch.serviceAccount .Values.global.serviceAccount | first }}
+{{- end -}}
+
+
 {{/*
 Default backend container security context.
 */}}
diff --git a/charts/nginx-ingress/templates/admission-webhooks/job-patch/serviceaccount.yaml b/charts/nginx-ingress/templates/admission-webhooks/job-patch/serviceaccount.yaml
index 52f94dcce..c5bed0614 100644
--- a/charts/nginx-ingress/templates/admission-webhooks/job-patch/serviceaccount.yaml
+++ b/charts/nginx-ingress/templates/admission-webhooks/job-patch/serviceaccount.yaml
@@ -13,5 +13,5 @@ metadata:
     {{- with .Values.controller.admissionWebhooks.patch.labels }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
-automountServiceAccountToken: {{ .Values.controller.admissionWebhooks.patch.serviceAccount.automountServiceAccountToken }}
+{{ include "ingress-nginx.admissionWebhooks.patch.automountServiceAccountToken" . }}
 {{- end }}
diff --git a/charts/nginx-ingress/templates/controller-serviceaccount.yaml b/charts/nginx-ingress/templates/controller-serviceaccount.yaml
index 48668374c..46e120e00 100644
--- a/charts/nginx-ingress/templates/controller-serviceaccount.yaml
+++ b/charts/nginx-ingress/templates/controller-serviceaccount.yaml
@@ -15,5 +15,5 @@ metadata:
   {{- if .Values.serviceAccount.annotations }}
   annotations: {{ toYaml .Values.serviceAccount.annotations | nindent 4 }}
   {{- end }}
-automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
+{{ include "ingress-nginx.automountServiceAccountToken" . }}
 {{- end }}
diff --git a/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml b/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
index 30170aa81..f023c054e 100644
--- a/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
+++ b/charts/nginx-ingress/templates/default-backend-serviceaccount.yaml
@@ -12,5 +12,5 @@ metadata:
     {{- end }}
   name: {{ include "ingress-nginx.defaultBackend.serviceAccountName" . }}
   namespace: {{ include "ingress-nginx.namespace" . }}
-automountServiceAccountToken: {{ .Values.defaultBackend.serviceAccount.automountServiceAccountToken }}
+{{ include "ingress-nginx.defaultBackend.automountServiceAccountToken" . }}
 {{- end }}
